text_localization_environment/TextLocEnv.py

```diff
@@ -177,11 +182,13 @@ def create_ior_mark(self):
 
     def compute_best_iou(self):
         max_iou = 0
+        i = -1
 
         for box in self.episode_true_bboxes:
             max_iou = max(max_iou, self.compute_iou(box))
+            i += 1
 
-        return max_iou
+        return max_iou, i
```

Comment:
Won’t this function return the size of self.episode_true_bboxes - 1 all the time, since i is increased during the for loop independent from whether the bounding box was larger than the previously largest box or not? 

My suggestion would be to change the for loop to something like:

```python
i += 1
candidate_iou = self.compute_iou(box)
if candidate_iou > max_iou:
  max_i = i
  max_iou = candidate_iou
```

and then return max_iou, max_i

Reply:
No worries … it's always hard to get something like this right from the beginning. Sometimes it helps to write some tests that formalize what one would expect as a result from the function. By executing those tests, one than can easily see whether something went wrong or everything works – or at least seems to. 🙂

Reply:
yeah tests are on the way.... I'm only missing the ones for the step function and then I will create the pull request